lib/_http_outgoing.js

```diff
@@ -90,11 +90,12 @@ exports.OutgoingMessage = OutgoingMessage;
 
 OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 
-  if (callback) {
-    if (typeof callback !== 'function')
-      throw new TypeError('callback must be a function');
+  if (typeof callback === 'function') {
     this.on('timeout', callback);
   }
+  else if (callback) {
+    throw new TypeError('callback must be a function');
```

Comment:
In other words, I think @thefourtheye is saying this would be fine:

```
if (callback) {
    this.on('timeout', callback);
}
```

Which throws a wrench in the whole "standardize on `typeof` checks" approach.